controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	err := r.get(rookCephOperatorConfig)
+	if err != nil {
+		r.log.Error(err, "Unable to get RookCeph ConfigMap.", "RookCephConfigMap", klog.KRef(r.storageCluster.Namespace, "rook-ceph-operator-config"))
+		return err
+	}
+	enableDisableFlagStr := fmt.Sprintf("%v", enableRookCSICephFS)
+
+	// If the current state of 'ROOK_CSI_ENABLE_CEPHFS' flag is same, just return.
+	if rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] == enableDisableFlagStr {
+		return nil
+	}
+	rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] = enableDisableFlagStr
+	return r.Client.Update(r.ctx, rookCephOperatorConfig)
+}
+
+func (r *StorageClassClaimReconciler) createStorageClasses(availableStorageClassConfigurations []StorageClassConfiguration, storageClassClaimOwnerReference metav1.OwnerReference) error {
+	var skippedStorageClass []string
+	for _, storageclassConfiguration := range availableStorageClassConfigurations {
+		if storageclassConfiguration.reconcileStrategy == "ignore" || storageclassConfiguration.disable {
+			continue
+		}
+
+		// StorageClass received in the response does not have the same name as the consumer-created StorageClassClaim.
+		storageClass := storageclassConfiguration.storageClass
+		if strings.Contains(storageClass.Name, "-ceph-rbd") && !storageclassConfiguration.isClusterExternal {
+
+			// Wait for CephBlockPool to be ready.
+			cephBlockPool := cephv1.CephBlockPool{}
+			err := r.get(&cephBlockPool)
+			if err != nil || cephBlockPool.Status == nil || cephBlockPool.Status.Phase != cephv1.ConditionType(util.PhaseReady) {
+				r.log.Info("Waiting for CephBlockPool to be Ready. Skip reconciling StorageClass", "CephBlockPool", klog.KRef(storageClass.Parameters["pool"], storageClass.Parameters["clusterID"]), "StorageClass", klog.KRef("", storageClass.Parameters["clusterID"]))
+				skippedStorageClass = append(skippedStorageClass, storageClass.Name)
+				continue
+			}
+		} else if strings.Contains(storageClass.Name, "-cephfs") && !storageclassConfiguration.isClusterExternal {
+
+			// Wait for CephFilesystem to be ready.
+			cephFilesystem := cephv1.CephFilesystem{}
+			err := r.get(&cephFilesystem)
+			if err != nil || cephFilesystem.Status == nil || cephFilesystem.Status.Phase != cephv1.ConditionType(util.PhaseReady) {
+				r.log.Info("Waiting for CephFilesystem to be Ready. Skip reconciling StorageClass", "CephFilesystem", klog.KRef(storageClass.Parameters["fsName"], storageClass.Parameters["clusterID"]), "StorageClass", klog.KRef("", storageClass.Name))
+				skippedStorageClass = append(skippedStorageClass, storageClass.Name)
+				continue
+			}
+		}
+
+		// Any existing (or newly created) StorageClass will have the same name as the consumer-created StorageClassClaim, which will also own it.
+		storageClass.Name = r.storageClassClaim.Name
+		storageClass.OwnerReferences = []metav1.OwnerReference{storageClassClaimOwnerReference}
```

Comment:
StorageClass is cluster scoped and Claim is namespaced. So claim can not be the owner of StorageClass.
Remove this ownerref.